azurerm/resource_arm_key_vault_certificate.go

```diff
@@ -549,13 +615,31 @@ func flattenKeyVaultCertificatePolicy(input *keyvault.CertificatePolicy) []inter
 		for _, usage := range *props.KeyUsage {
 			usages = append(usages, string(usage))
 		}
+
+		sanOutput := make(map[string]interface{}, 0)
+		if san := props.SubjectAlternativeNames; san != nil {
+			if emails := san.Emails; emails != nil {
+				sanOutput["emails"] = *emails
+			}
+
+			if dnsNames := san.DNSNames; dnsNames != nil {
+				sanOutput["dns_names"] = *dnsNames
+			}
+
+			if upns := san.Upns; upns != nil {
+				sanOutput["upns"] = *upns
+			}
+		}
+
 		certProps["key_usage"] = usages
 		certProps["subject"] = *props.Subject
 		certProps["validity_in_months"] = int(*props.ValidityInMonths)
 		if props.Ekus != nil {
 			certProps["extended_key_usage"] = props.Ekus
 		}
-
+		if len(sanOutput) > 0 {
```

Comment:
these changes above allow us to ensure that `certProps["subject_alternative_names"]` is set either case (either to an empty list or to a list containing the item (as here). as such (with the change above) we should be able to make this:

```suggestion
					certProps["subject_alternative_names"] = sanOutputs
```